src/section/section_keys.rs

```diff
@@ -38,13 +39,25 @@ pub struct SectionKeysProvider {
 }
 
 impl SectionKeysProvider {
+    pub fn has_key_or_candidate(&self) -> bool {
+        self.current.is_some() || !self.new.is_empty()
```

Comment:
Is this `|| !self.new.is_empty()` valid in this sense, could it have content but not the content we wish?

Reply:
> could it have content but not the content we wish

There is check when populating this field, and this field will be clear once got updated properly.
So, I think we don't need to carry a further check here.

Reply:
If the populated field is immutable then I think this is fine, but if it can be mutated, perhaps we can look a bit further? 
i.e. I think correct and immutable on construction is perfectly valid and then this all is valid as well if that makes sense?

Reply:
Ok, will have a check.